diff --git a/rules.mk b/rules.mk
index 0e9c178c1b..78d1de2d47 100644
--- a/rules.mk
+++ b/rules.mk
@@ -493,8 +493,8 @@ endef
 # @param 2: If set, recurse into subdirectories.
 ##
 define sha256sums
-	(cd $(1); find . $(if $(2),,-maxdepth 1) -type f -not -name 'sha256sums' -printf "%P\n" | sort | \
-		xargs -r $(MKHASH) -n sha256 | sed -ne 's!^\(.*\) \(.*\)$$!\1 *\2!p' > sha256sums)
+	(cd $(1); find . $(if $(2),,-maxdepth 1) -type f -not -name 'sha256sums' -printf "%P\0" | sort -z | \
+		xargs -0 -r $(MKHASH) -n sha256 | sed -ne 's!^\(.*\) \(.*\)$$!\1 *\2!p' > sha256sums)
 endef
 
 ##@
